src/utils/global_reentrancy_guard.cairo

```diff
@@ -8,9 +8,16 @@ use starknet::ContractAddress;
 use satoru::data::data_store::{IDataStoreDispatcher};
 
 /// Modifier to avoid reentrancy.
-fn non_reentrant_before(data_store: IDataStoreDispatcher) { // TODO
+fn non_reentrant_before(data_store: IDataStoreDispatcher) { 
+    let status = data_store.get_bool('REENTRANCY_GUARD_STATUS'); // Read key from Data Store
+
+    assert(!status, 'ReentrancyGuard: reentrant call');
```

Comment:
nit: error message in a constant in an error file, following conventions of other error messages